### ISSUE TITLE
TinyImageNet Dataset Performance

### ISSUE THREAD

**Issue by LKONE-1019** [nazmul-karim170/UNICON-Noisy-Label#6]
Hello. I have run your repository directly over Tiny-ImageNet dataset to reproduce experimental results. The parameter setting is consistent with the following parameter table:
![微信截图_20220723223324](https://user-images.githubusercontent.com/77560173/180610337-69df2f5f-27f8-4464-af74-401f82742518.png)
The specific parameter settings are as follows:
```
parser = argparse.ArgumentParser(description='PyTorch Clothing1M Training')
parser.add_argument('--batch_size', default=32, type=int, help='train batchsize')
parser.add_argument('--lr', '--learning_rate', default=0.01, type=float, help='initial learning rate')
parser.add_argument('--alpha', default=2, type=float, help='parameter for Beta')
parser.add_argument('--lambda_u', default=30, type=float, help='weight for unsupervised loss')
parser.add_argument('--lambda_c', default=0.025, type=float, help='weight for contrastive loss')
parser.add_argument('--T', default=0.5, type=float, help='sharpening temperature')
parser.add_argument('--num_epochs', default=350, type=int)
parser.add_argument('--id', default='TinyImage')
parser.add_argument('--data_path', default='./data/tiny-imagenet-200', type=str, help='path to dataset')
parser.add_argument('--seed', default=123)
parser.add_argument('--gpuid', default=5, type=int)
parser.add_argument('--noise_mode',  default='sym')
parser.add_argument('--d_u',  default=0.7, type=float)
parser.add_argument('--tau',  default=5, type=float)
parser.add_argument('--ratio', default=0.2 , type=float, help='noise ratio')
parser.add_argument('--resume', default=False , type=bool, help='Resume from chekpoint')
parser.add_argument('--num_class', default=200, type=int)
parser.add_argument('--num_batches', default=2000, type=int)
parser.add_argument('--dataset', default='TinyImageNet', type=str)
```
```
## Loss Functions and Optimizers 
criterion = SemiLoss()
optimizer1 = optim.SGD(net1.parameters(), lr=args.lr, momentum=0.9, weight_decay=5e-4)
optimizer2 = optim.SGD(net2.parameters(), lr=args.lr, momentum=0.9, weight_decay=5e-4)
```

I have run Train_TinyImageNet.py train script with following command:
```
python Train_TinyImageNet.py --gpuid 3 --ratio 0
```
```
python Train_TinyImageNet.py --gpuid 4 --ratio 0.2
```
```
python Train_TinyImageNet.py --gpuid 5 --ratio 0.5
```
The final accuracy under the three noise rates are:
```
Number of Samples: 100000
Train Net1
labeled data has a size of 52228
unlabeled data has a size of 47800
TinyImageNet:0.0-sym | Epoch [350/350] Iter[1632/1633]   Labeled loss: 2.68  Unlabeled loss: 0.00 Contrastive Loss:0.0055Train Net2
labeled data has a size of 50222
unlabeled data has a size of 49800
TinyImageNet:0.0-sym | Epoch [350/350] Iter[1569/1570]   Labeled loss: 2.68  Unlabeled loss: 0.00 Contrastive Loss:0.0053
| Test Acc: 51.49%
```
```
Number of Samples: 100000
Train Net1
labeled data has a size of 47362
unlabeled data has a size of 52800
TinyImageNet:0.2-sym | Epoch [350/350] Iter[1480/1481]   Labeled loss: 2.74  Unlabeled loss: 0.00 Contrastive Loss:0.0050Train Net2
labeled data has a size of 45690
unlabeled data has a size of 54400
TinyImageNet:0.2-sym | Epoch [350/350] Iter[1427/1428]   Labeled loss: 2.73  Unlabeled loss: 0.00 Contrastive Loss:0.0053
| Test Acc: 50.49%
```
```
Number of Samples: 100000
Train Net1
labeled data has a size of 36545
unlabeled data has a size of 63600
TinyImageNet:0.5-sym | Epoch [350/350] Iter[1142/1143]   Labeled loss: 2.77  Unlabeled loss: 0.00 Contrastive Loss:0.0061Train Net2
labeled data has a size of 36014
unlabeled data has a size of 64000
TinyImageNet:0.5-sym | Epoch [350/350] Iter[1125/1126]   Labeled loss: 2.82  Unlabeled loss: 0.00 Contrastive Loss:0.0063
| Test Acc: 47.16%
```
which is quite different from the results shown in the paper. Is there something wrong with the experimental setup？
Hope you can give some suggestions. Thank you!

**Comment by nazmul-karim170** [comment]
I will look into this. These results does not look right. Can you change
the learning rate?

**Comment by nazmul-karim170** [comment]
I will get back to you with proper hyper parameters (this is the issue).

Thanks,
Nazmul

**Comment by nazmul-karim170** [comment]
Can you run this Command-
      python Train_TinyImageNet.py --ratio 0.2 

Please change the ratio accordingly. The issue was with the learning rate and its decay rate.

**Comment by LKONE-1019** [comment]
- Is the hyperparameter in the script now aimed at noise rate 0.2?

- I ran the code you gave two days ago directly, but the result didn't seem to change now.

![code](https://user-images.githubusercontent.com/77560173/181262651-9113cfce-045b-459e-b824-fb9729e5bae0.png)

- I will run your new code, thank you~

**Comment by nazmul-karim170** [comment]
You can run them with any noise rate.
